File: python/mt19973.py
import numpy.random as vrng
from sys import argv

N = int(argv[1]) if len(argv) > 1 else 10
seed = int(argv[2]) if len(argv) > 2 else 1234

mt19937 = vrng.RandomState(seed)

def rand_uint32():
    return mt19937.randint(0,2**32)

for _ in range(N):
    print(rand_uint32())

File: c++/mt19973.cpp
#include <ctime>
#include <iostream>
#include <random>
  
int main(int argc, char *argv[])
{
  int N = (argc>1) ? atoi(argv[1]) : 10;
  int seed = (argc>2) ? atoi(argv[2]) : 1234;
  std::mt19937 rand_uint32(seed); 
 
  for(; N>0; --N) { 
    std::cout << rand_uint32() << std::endl; 
  }

  return 0;
}
